HLTrigger/special/plugins/HLTMuonRechitClusterFilter.cc

```diff
@@ -0,0 +1,113 @@
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/maker/WorkerMaker.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+class HLTMuonRechitClusterFilter : public edm::global::EDFilter<> {
+public:
+  explicit HLTMuonRechitClusterFilter(const edm::ParameterSet&);
+  ~HLTMuonRechitClusterFilter() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  bool filter(edm::StreamID, edm::Event&, edm::EventSetup const&) const override;
+  const edm::InputTag cluster_tag_;
+  const edm::EDGetTokenT<reco::MuonRecHitClusterCollection> cluster_token_;
+  const int min_N_;
+  const int min_Size_;
+  const int min_SizeMinusMB1_;
+  const int max_nMB1_;
+  const int max_nMB2_;
+  const int max_nME11_;
+  const int max_nME12_;
+  const int max_nME41_;
+  const int max_nME42_;
+  const int min_nStation_;
+  const double min_avgStation_;
+  const double min_Time_;
+  const double max_Time_;
+  const double min_Eta_;
+  const double max_Eta_;
+  const double max_TimeSpread_;
+};
+//
+// constructors and destructor
+//
+HLTMuonRechitClusterFilter::HLTMuonRechitClusterFilter(const edm::ParameterSet& iConfig)
+    : cluster_tag_(iConfig.getParameter<edm::InputTag>("ClusterTag")),
+      cluster_token_(consumes<reco::MuonRecHitClusterCollection>(cluster_tag_)),
+      min_N_(iConfig.getParameter<int>("MinN")),
+      min_Size_(iConfig.getParameter<int>("Min_Size")),
+      min_SizeMinusMB1_(iConfig.getParameter<int>("Min_SizeMinusMB1")),
+      max_nMB1_(iConfig.getParameter<int>("Max_nMB1")),
+      max_nMB2_(iConfig.getParameter<int>("Max_nMB2")),
+      max_nME11_(iConfig.getParameter<int>("Max_nME11")),
+      max_nME12_(iConfig.getParameter<int>("Max_nME12")),
+      max_nME41_(iConfig.getParameter<int>("Max_nME41")),
+      max_nME42_(iConfig.getParameter<int>("Max_nME42")),
+      min_nStation_(iConfig.getParameter<int>("Min_nStation")),
+      min_avgStation_(iConfig.getParameter<double>("Min_avgStation")),
+      min_Time_(iConfig.getParameter<double>("Min_Time")),
+      max_Time_(iConfig.getParameter<double>("Max_Time")),
+      min_Eta_(iConfig.getParameter<double>("Min_Eta")),
+      max_Eta_(iConfig.getParameter<double>("Max_Eta")),
+      max_TimeSpread_(iConfig.getParameter<double>("Max_TimeSpread")) {}
+
+void HLTMuonRechitClusterFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("ClusterTag", edm::InputTag("hltCSCrechitClusters"));
+  desc.add<int>("MinN", 1);
+  desc.add<int>("Min_Size", 50);
```

Comment:
```suggestion
  desc.add<int>("MinN", 1);
  desc.add<int>("MinSize", 50);
```
Minor, but I would suggest to be consistent with the use of `_` (`MinN` and `Min_Size` are inconsistent). I would go without underscores. This would then also apply to the class members (e.g. `minSize_`).

Reply:
@missirol I was following the example here:
https://cmssdt.cern.ch/lxr/source/HLTrigger/HLTfilters/plugins/HLTSinglet.cc?v=CMSSW_11_2_1
where the class members have `_` (`min_N`) and config parameters use camel case (`MinN`).  
The exception is that I use lower case `n` before the station counters, and added `_` in front for readability (e.g. `Max_nMB1`).

How about using the following names?
```
      min_N_(iConfig.getParameter<int>("MinN")),
      min_Size_(iConfig.getParameter<int>("MinSize")),
      min_SizeMinusMB1_(iConfig.getParameter<int>("MinSizeMinusMB1")),
      max_nMB1_(iConfig.getParameter<int>("Max_nMB1")),
      max_nMB2_(iConfig.getParameter<int>("Max_nMB2")),
      max_nME11_(iConfig.getParameter<int>("Max_nME11")),
      max_nME12_(iConfig.getParameter<int>("Max_nME12")),
      max_nME41_(iConfig.getParameter<int>("Max_nME41")),
      max_nME42_(iConfig.getParameter<int>("Max_nME42")),
      min_Nstation_(iConfig.getParameter<int>("MinNStation")),
      min_AvgStation_(iConfig.getParameter<double>("MinAvgStation")),
      min_Time_(iConfig.getParameter<double>("MinTime")),
      max_Time_(iConfig.getParameter<double>("MaxTime")),
      min_Eta_(iConfig.getParameter<double>("MinEta")),
      max_Eta_(iConfig.getParameter<double>("MaxEta")),
      max_TimeSpread_(iConfig.getParameter<double>("MaxTimeSpread"))
```

Reply:
That's okay with me; thanks for the clarification.